### ISSUE TITLE
Curve intersection fails when non-default start / end are used

### ISSUE THREAD

**Issue by dhermes** [dhermes/bezier#44]
For example:

```python
>>> import numpy as np
>>> import bezier
>>> from bezier._intersection_helpers import all_intersections
>>>
>>> nodes15 = np.asfortranarray([
...     [0.25 , 0.625],
...     [0.625, 0.25 ],
...     [1.0  , 1.0  ],
])
>>> nodes25 = np.asfortranarray([
...     [0.0 , 0.5],
...     [0.25, 1.0],
...     [0.75, 1.5],
...     [1.0 , 0.5],
... ])
>>>
>>> curve15a = bezier.Curve(nodes15, degree=2)
>>> curve25a = bezier.Curve(nodes25, degree=3)
>>> intersection, = all_intersections([(curve15a, curve25a)])
>>> intersection.s
0.8587897065534015
>>> intersection.t
0.8734528541508781
>>> intersection.get_point()
array([[ 0.89409228,  0.81061745]])
```

but if we just use non-default `start` and `end` this fails:

```python
>>> curve15b = bezier.Curve(nodes15, degree=2, start=0.125, end=0.75)
>>> curve25b = bezier.Curve(nodes25, degree=3, start=0.25 , end=2.0)
>>> all_intersections([(curve15b, curve25b)])
all_intersections([(curve15b, curve25b)])
Traceback (most recent call last):
...
ValueError: outside of unit interval
```

even though the answer should be exactly the same:

```python
>>> curve15b.evaluate(intersection.s)
array([[ 0.89409228,  0.81061745]])
>>> curve25b.evaluate(intersection.t)
array([[ 0.89409228,  0.81061745]])
```

----

At 8271bf87567278e8d4b90913e0cef418cacb1181

**Comment by dhermes** [comment]
The **real** fix is to actually remove `_start` and `_end` from `Curve`. As seen in #43, the `start` and `end` value only matters **relative** to the computation, so it should be passed around in that context.

This is a positive anyhow: passing around `Curve` instances does not lend itself to ease of implementation in Fortran.

**Comment by dhermes** [comment]
This has been fixed as of db427f958266103e0a266721e18c66a7025d85ae.

This issue was fixed a few commits before, I'm just not sure which one fixed it. The commit where `_geometric_intersection.all_intersections` switched from receiving a `Curve` to just raw nodes is the one that fixed it.
